neurom/core/soma.py

```diff
@@ -149,6 +158,29 @@ def __str__(self):
         return ('SomaCylinders(%s) <center: %s, virtual radius: %s>' %
                 (repr(self.points), self.center, self.radius))
 
+    def overlaps(self, points, exclude_boundary=False):
+        """Check that the given points are located inside the soma."""
+        points = np.atleast_2d(np.asarray(points, dtype=np.float64))
+        mask = np.ones(len(points)).astype(bool)
+        for p1, p2 in zip(self.points[:-1], self.points[1:]):
+            vec = p2[COLS.XYZ] - p1[COLS.XYZ]
+            vec_norm = np.linalg.norm(vec)
+            dot = np.einsum("i,ji", vec, points[mask] - p1[COLS.XYZ]) / vec_norm
```

Comment:
isnt' this faster? (not sure) 

```(points[mask] - p1[COLS.XYZ]).dot(vec)```

Reply:
Indeed, and much more readable :)

Reply:
not sure exactly what you did, but I just did this cylinder check a few days ago, you want to do this:
if cylinder is parametrized by pos_vec + x * dir_vec (2 vectors and parameter x) for the centerline and radius r, the distance from a point p to centerline of cylinder is `np.linalg.norm(np.cross(pos_vec-p, dir_vec))`, which you can just check against the radius. Maybe this is what you did ;)